### ISSUE TITLE
Plugin restarted during E2E parallel volume create tests

### ISSUE THREAD

**Issue by govint** [vmware/docker-volume-vsphere#1395]
CI (https://ci.vmware.run/vmware/docker-volume-vsphere/464) failed with below error,
FAIL: volumecreate_test.go:177: VolumeCreateTestSuite.TestValidOptions

volumecreate_test.go:82:
```
// positive test case
c.Assert(err, IsNil, Commentf(out))
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc420363140), Stderr:[]uint8(nil)} ("exit status 1")
... Error response from daemon: create option_volume_n9ocblmgyv6t: Post http://%!F(MISSING)run%!F(MISSING)docker%!F(MISSING)plugins%!F(MISSING)334cbf101410e47e1360d8793088a0e9b8fe25e263fe87cbe190729a8020203c%!F(MISSING)vsphere.sock/VolumeDriver.Create: http: ContentLength=69 with Body length 0
```

volumecreate_test.go:82:
```
// positive test case
c.Assert(err, IsNil, Commentf(out))
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc42045d360), Stderr:[]uint8(nil)} ("exit status 1")
... Error response from daemon: create option_volume_1x211hubfnfyi: Post http://%!F(MISSING)run%!F(MISSING)docker%!F(MISSING)plugins%!F(MISSING)334cbf101410e47e1360d8793088a0e9b8fe25e263fe87cbe190729a8020203c%!F(MISSING)vsphere.sock/VolumeDriver.Create: http: ContentLength=80 with Body length 0
```
From the test logs looks like the plugin restarted around the time the volume create req was made:
```
2017/06/12 14:03:35 Creating volume [option_volume_n9ocblmgyv6t] with options [ -o access=read-write] on VM [192.168.31.241]
2017/06/12 14:03:56 Failed to invoke command [docker volume  create --driver=vsphere --name=option_volume_n9ocblmgyv6t  -o access=read-write]: exit status 1
```
ANd plugin logs around this time shows a restart,

```
2017-06-12 14:03:41.809238398 +0000 UTC [INFO] Scan complete device="/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0" event="/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0": 0x100 == IN_CREATE 
```
Plugin has restarted here,
```
2017-06-12 14:03:42.189414369 +0000 UTC [INFO] No config file found. Using defaults.
2017-06-12 14:03:42.189457766 +0000 UTC [WARNING] Failed to load config file /etc/docker-volume-vsphere.conf: open /etc/docker-volume-vsphere.conf: no such file or directory
2017-06-12 14:03:42.189477992 +0000 UTC [INFO] Starting plugin config="/etc/docker-volume-vsphere.conf" driver=vsphere log_level=info 
2017-06-12 14:03:42.189489074 +0000 UTC [INFO] Plugin options - port=1019 
2017-06-12 14:03:42.189499786 +0000 UTC [INFO] Getting volume data from unix:///var/run/docker.sock
2017-06-12 14:03:42.204896214 +0000 UTC [INFO] Found 0 running or paused containers
```
May be skip this test and lets run it outside of CI to see if the issue is repro'ed.

**Comment by pdhamdhere** [comment]
Looks like `/var/log/syslog` is empty for VM1 192.168.31.241 CC/ @shuklanirdesh82 

Not sure how we can debug this further in absence of logs. Issue doesn't seem to be 100% reproducible either. Close?

**Comment by govint** [comment]
Issue repro'ed again in CI - https://ci.vmware.run/vmware/docker-volume-vsphere/490 (PR #1383 CI was failed because of this)

Seems like the clone volume test is able to get the plugin to restart.
---
2017-06-13 09:53:24.86751517 +0000 UTC [INFO] Attaching volume and creating filesystem name="clone_src_volume_59anldo9p9od" fstype=ext4 
2017-06-13 09:53:25.564749226 +0000 UTC [INFO] Scan complete device="/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0" event="/dev/disk/by-path/pci-0000:0b:00.0-scsi-0:0:0:0": 0x100 == IN_CREATE 


2017-06-13 09:53:26.246796104 +0000 UTC [INFO] No config file found. Using defaults.
2017-06-13 09:53:26.246968474 +0000 UTC [WARNING] Failed to load config file /etc/docker-volume-vsphere.conf: open /etc/docker-volume-vsphere.conf: no such file or directory
2017-06-13 09:53:26.246982139 +0000 UTC [INFO] Starting plugin driver=vsphere log_level=info config="/etc/docker-volume-vsphere.conf" 
2017-06-13 09:53:26.247020671 +0000 UTC [INFO] Plugin options - port=1019 
2017-06-13 09:53:26.247033976 +0000 UTC [INFO] Getting volume data from unix:///var/run/docker.sock
2017-06-13 09:53:26.260895897 +0000 UTC [INFO] Found 0 running or paused containers
2017-06-13 09:53:26.261167693 +0000 UTC [INFO] Discovered 0 volumes in use.
2017-06-13 09:53:26.26117792 +0000 UTC [INFO] Refcounting successfully completed

**Comment by govint** [comment]
Last time this issue was resolved (used to crash in the same function) by moving up to a newer GO runtime.

@shuklanirdesh82 can you say whats the GO run time on the photon VMs used for this test?

**Comment by shuklanirdesh82** [comment]
> Looks like /var/log/syslog is empty for VM1 192.168.31.241

Our script does collect syslogs .. need to check upon vm to find out what has changed.

> can you say whats the GO run time on the photon VMs used for this test?

I believe go1.7; AFAIK we have upgraded couple months back.

**Comment by govint** [comment]
@shuklanirdesh82 to debug this may need to be able to get access to the VM after the test if thats possible. Locally it seems to be working ok though.

**Comment by shuklanirdesh82** [comment]
> https://github.com/vmware/docker-volume-vsphere/issues/1395#issuecomment-308185895

Aren't logs enough? Let me know what is missing.

**Comment by govint** [comment]
Logs aren't enough if we need to catch the plugin when it restarts - that can happen only if we can control the execution of the plugin and be able to get a dump or better debug with a debugger at least get some data on the state of the plugin (thread state). Else there's no way to debug this further.

**Comment by pdhamdhere** [comment]
Same failure: https://ci.vmware.run/vmware/docker-volume-vsphere/527

Syslog for Photon VM is missing. We should find some hints in syslog in case of plugin crash.

**Comment by govint** [comment]
These tests that show the issue aren't enabled and hence not blocking any CI tests. Plus #1432 which has reenabled the tests isn't repro'ing the issue.

**Comment by pdhamdhere** [comment]
If issue is not longer reproducible, let's close the issue.

**Comment by shuklanirdesh82** [comment]
> #1432 which has reenabled the tests isn't repro'ing the issue.
> https://github.com/vmware/docker-volume-vsphere/issues/1395#issuecomment-309652305

Reopening this issue to re-enable the test.

**Comment by shuklanirdesh82** [comment]
fixed by #1461
